data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/ExpiryMonitor.java

```diff
@@ -0,0 +1,48 @@
+package org.opensearch.dataprepper.acknowledgements;
+
+import org.opensearch.dataprepper.model.acknowledgements.ExpiryItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class ExpiryMonitor {
+    private static final Logger LOG = LoggerFactory.getLogger(ExpiryMonitor.class);
+
+    private static final long CANCEL_CHECK_INTERVAL_SECONDS = 15;
+    private static final ScheduledExecutorService SCHEDULED_EXECUTOR_SERVICE = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Let's do something like this:

https://github.com/opensearch-project/data-prepper/blob/cdf404e0b464320f34582e2259d6c3868384199f/data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/source/RandomStringSource.java#L37-L39

This will yield a clearer name for the threads which helps during debugging. And the non-daemon thread will help Data Prepper shutdown correctly.